### PR TITLE
Init

### PR DESCRIPTION
**Pull request by bryanloh-jw**
Initialized the frontend portion of the application, which uses react with typescript.

Have created the template for the landing page, still a work in progress but the main theme and look has been set.

### CODE REVIEW

File: the-name-card/src/main.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.scss';
+import { Landing, Margin } from './components';
+
+ReactDOM.createRoot(document.getElementById('root')!).render(
+	<React.StrictMode>
+		<Margin>
+			<Landing />
```
Comment (raynerljm) [inline the-name-card/src/main.tsx]: you should be rendering `App.tsx` here, not `Landing`, your `Margin` and `Landing` components should be in `App` instead

File: the-name-card/src/components/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export { default as Landing } from './Landing/Landing';
```
Comment (raynerljm) [inline the-name-card/src/components/index.ts]: nit: personally i prefer using named exports 

e.g.
```
// Section.tsx
const Section ...
...

export { Section };
```

```
// components/index.ts

export * from './Section/Section'
```

File: the-name-card/src/components/Section/Section.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React, { ReactNode } from 'react';
+import './Section.scss';
+
+interface SectionProps {
+	children: ReactNode;
```
Comment (raynerljm) [inline the-name-card/src/components/Section/Section.tsx]: this is 1 valid way to do this

here are some other ways
https://stackoverflow.com/questions/71788254/react-18-typescript-children-fc
https://stackoverflow.com/questions/71788254/react-18-typescript-children-fc/71800185#71800185

File: the-name-card/src/components/Margin/Margin.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { FC, PropsWithChildren, ReactNode } from 'react';
+import './Margin.scss';
+
+interface MarginProps {
+	children: ReactNode | null;
```
Comment (raynerljm) [inline the-name-card/src/components/Margin/Margin.tsx]:
```suggestion
	children?: ReactNode;
```

File: the-name-card/src/components/CardHolder/CardHolder.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { FC, PropsWithChildren, ReactNode } from 'react';
+import { Section } from '..';
+import './CardHolder.scss';
+
+interface CardHolderProps {
+	children: ReactNode | ReactNode[];
+}
+
+const CardHolder: FC<PropsWithChildren<CardHolderProps>> = ({ children }) => {
```
Comment (raynerljm) [inline the-name-card/src/components/CardHolder/CardHolder.tsx]: don't have to use `PropsWithChildren` wrapper if you are already defining the `children` attribute in the props

also good to stick to one convention for the entire project

File: the-name-card/src/components/Card/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import './Card.scss';
+import imageUrl from './../../assets/svgs/undraw_portfolio_update_re_jqnp.svg';
+
+const details = {
```
Comment (raynerljm) [inline the-name-card/src/components/Card/Card.tsx]: assuming this is dummy placeholder data first right - good to add a comment just to note

**Review by raynerljm — APPROVED** [review]
left some comments but mostly nits

File: the-name-card/src/main.tsx (hunk 1 above)
Comment (bryanloh-jw) [reply the-name-card/src/main.tsx]: yeap mb, should be putting the navigation in for this. Will add that in before resolving this

File: the-name-card/src/components/index.ts (hunk 2 above)
Comment (bryanloh-jw) [reply the-name-card/src/components/index.ts]: right that is good suggestion

**Pull request merged**